src/liballoc/boxed.rs

```diff
@@ -623,6 +624,66 @@ impl From<Box<str>> for Box<[u8]> {
     }
 }
 
+/// The error type returned when a conversion from a boxed slice to a boxed
+/// array fails.
+#[unstable(feature = "try_from", issue = "33417")]
+#[derive(Debug, Copy, Clone)]
+pub struct TryFromSliceError<T>(Box<[T]>);
+
+impl<T> fmt::Display for TryFromSliceError<T> {
+    #[inline]
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        fmt::Display::fmt(self.__description(), f)
+    }
+}
+
+impl<T> TryFromSliceError<T> {
+    #[unstable(feature = "array_error_internals",
+           reason = "available through Error trait and this method should not \
+                     be exposed publicly",
+           issue = "0")]
+    #[inline]
+    #[doc(hidden)]
+    pub fn __description(&self) -> &str {
+        "could not convert boxed slice to boxed array"
+    }
+
+    /// Returns the boxed slice that was attempted to convert to a boxed array.
+    ///
+    /// This method is meant to avoid allocation. It will consume the error,
+    /// moving out the boxed slice, so that a copy of the slice does not need to
+    /// be made.
+    #[inline]
+    pub fn into_boxed_slice(self) -> Box<[T]> {
+        self.0
+    }
+}
+
+macro_rules! array_impls {
+    ($($N:expr)+) => {
```

Comment:
You forgot to add `$(...)+` around the item.

```
[00:05:12]    Compiling alloc v0.0.0 (file:///checkout/src/liballoc)
[00:05:12] error: variable 'N' is still repeating at this depth
[00:05:12]    --> liballoc/boxed.rs:665:47
[00:05:12]     |
[00:05:12] 665 |         impl<T> TryFrom<Box<[T]>> for Box<[T; $N]> {
[00:05:12]     |                                               ^^
[00:05:12] 
[00:05:12] error: Could not compile `alloc`.
```

Reply:
I add the current state of the master branch and idk why all of that is there. I'll reset back to the initial commit tomorrow.